lgc/state/PalMetadata.cpp

```diff
@@ -969,3 +920,178 @@ void PalMetadata::eraseFragmentInputInfo() {
+  if (getNumberOfSgprsBeforeUserData(callingConv) == 0)
+    return 10;
+  return 0;
+}
+
+// =====================================================================================================================
+// Returns the number of SGPRs that contain parameters for the shader after the user data.
+//
+// @param callingConv : The calling convention of the shader stage
+//
+unsigned PalMetadata::getVertexIdOffset(unsigned callingConv) {
+  switch (callingConv) {
+  case CallingConv::AMDGPU_LS: // Before-GFX9 unmerged LS
+  case CallingConv::AMDGPU_ES: // Before-GFX9 unmerged ES
+  case CallingConv::AMDGPU_VS:
+    return 0;
+  case CallingConv::AMDGPU_HS: // GFX9+ LS+HS
+    return 2;
+  case CallingConv::AMDGPU_GS: // GFX9+ ES+GS
+    return 5;
+  default:
+    llvm_unreachable("Unexpected calling convention.");
+    return UINT_MAX;
+  }
+}
+
+// =====================================================================================================================
+// Returns the offset of the register containing the instance id in the shader.
+//
+// @param callingConv : The calling convention of the shader stage
+//
+unsigned PalMetadata::getInstanceIdOffset(unsigned callingConv) {
+  switch (callingConv) {
+  case CallingConv::AMDGPU_LS: // Before-GFX9 unmerged LS
+  case CallingConv::AMDGPU_ES: // Before-GFX9 unmerged ES
+  case CallingConv::AMDGPU_VS:
+    return 3;
+  case CallingConv::AMDGPU_HS: // GFX9+ LS+HS
+    return 5;
+  case CallingConv::AMDGPU_GS: // GFX9+ ES+GS
+    return 8;
+  default:
+    llvm_unreachable("Unexpected calling convention.");
+    return UINT_MAX;
+  }
+}
+
+// =====================================================================================================================
+// Returns the number of VGPRs used for inputs to the shader.
+//
+// @param callingConv : The calling convention of the shader stage
+//
+unsigned PalMetadata::getVgprCount(unsigned callingConv) {
+  switch (callingConv) {
+  case CallingConv::AMDGPU_LS: // Before-GFX9 unmerged LS
+  case CallingConv::AMDGPU_ES: // Before-GFX9 unmerged ES
+  case CallingConv::AMDGPU_VS:
+    return 4;
+  case CallingConv::AMDGPU_HS: // GFX9+ LS+HS
+    return 6;
+  case CallingConv::AMDGPU_GS: // GFX9+ ES+GS
+    return 9;
+  default:
+    llvm_unreachable("Unexpected calling convention.");
+    return UINT_MAX;
+  }
+}
+
+// =====================================================================================================================
+// Returns true if the shader runs in wave32 mode.
+//
+// @param callingConv : The calling convention of the shader stage
+//
+bool PalMetadata::isWave32(unsigned callingConv) {
+  if (m_pipelineState->getTargetInfo().getGfxIpVersion().major < 10)
+    return false;
+
+  VGT_SHADER_STAGES_EN vgtShaderStagesEn;
+  vgtShaderStagesEn.u32All = m_registers[m_document->getNode(mmVGT_SHADER_STAGES_EN)].getUInt();
+  switch (callingConv) {
+  case CallingConv::AMDGPU_LS: // Before-GFX9 unmerged LS
```

Comment:
No need for these two cases as this bit of code is reached only for gfx10+.

Reply:
done.